src/assign.cpp

```diff
@@ -59,6 +60,26 @@ fba::Assignment::Assignment(fba::Targets::pshr tgs,
                 nassign_petal[tp][tile_id][p] = 0;
             }
             tile_target_xy[tile_id].clear();
+            if (tp == TARGET_TYPE_SKY) {
+                // for any stuck positioners that land on good sky,
+                // increment the counter
+                if (stuck_sky.count(tile_id)==0)
+                    continue;
+                for (auto & st : stuck_sky[tile_id]) {
+                    // st: < loc_id, bool >
+                    int32_t loc = st.first;
+                    bool good_sky = st.second;
+                    if (!good_sky)
+                        continue;
+                    int32_t petal = hw_->loc_petal[loc];
+                    nassign_tile .at(tp).at(tile_id)++;
+                    nassign_petal.at(tp).at(tile_id).at(petal)++;
+                    logmsg.str("");
```

Comment:
It may not matter here, but when printing debug messages from C++ inside nested loops, I found that string manipulation has a large performance impact.  That was the reason behind the pattern I'm sure you have seen elsewhere in the code:
```
bool extra_log = logger.extra_debug();

<snip>

if (extra_log) {
    logmsg.str("");

    <snip>

    logger.debug_tfg(...);
}
```
Which basically just checks a `bool` and cuts off the strstream manipulation.

Reply:
Thanks, done in 2221129